### PR TITLE
Ignore Unittests subtests for DISABLED_LEGACY_ENGINE

### PR DESCRIPTION
**Pull request by Shreeshrii**

### CODE REVIEW

**Comment by Shreeshrii** [comment]
Fixes https://github.com/tesseract-ocr/tesseract/issues/3226

**Comment by Shreeshrii** [comment]
@amitdo Are these changes ok or should I just exclude them in the makefile?

File: unittest/resultiterator_test.cc [hunk 1]
```diff
@@ -274,6 +274,10 @@ class ResultIteratorTest : public testing::Test {
 
 // Tests that Tesseract gets exactly the right answer on phototest.
 TEST_F(ResultIteratorTest, EasyTest) {
+#ifdef DISABLED_LEGACY_ENGINE
```
Comment (amitdo) [inline unittest/resultiterator_test.cc:277]: This test is a layout analysis test which does not really need osd.

It should work with lstm, if the oem and psm are adapted.

**Comment by amitdo** [comment]
I written a comment but removed it because I was not sure about it.

Do what you see as appropriate.

I don't have time to review it right now, sorry.

**Comment by Shreeshrii** [comment]
Ok. I will test with an older version to see if the tests were passing earlier.

**Comment by amitdo** [comment]
The changes look fine. In the iterator test we probably can re-enable some of the disabled tests with some additional work, but it's a good start.

**Comment by amitdo** [comment]
#3626

@Shreeshrii, do you remember if you tested the modified resultiterator_test.cc with --disable-legacy?

**Comment by amitdo** [comment]
@Shreeshrii, do you remember if you tested the modified resultiterator_test.cc with --disable-legacy?

**Comment by Shreeshrii** [comment]
No, I haven't tested recently. The CI for this was on a monthly basis. You can change it to daily similar to the other Actions.

**Comment by amitdo** [comment]
>No, I haven't tested recently.

But did you tested it when you originally submitted the PR? 

I already change it to run every Monday.

**Pull request closed without merging**